app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -21,6 +26,69 @@ const application = computed(() => {
     info?.methodCall.notAppliedArguments ?? [],
   )
 })
+
+const selfArgumentAstId = computed<Opt<ExprId>>(() => {
+  const tree = props.input
+  if (tree.isTree(Tree.Type.OprApp)) {
+    return tree.tryMap((tree) => tree.lhs)?.astId
+  } else if (tree.isTree(Tree.Type.App)) {
+    return tree.tryMap((tree) => tree.arg)?.astId
+  } else if (tree.isTree(Tree.Type.NamedApp)) {
+    return tree.tryMap((tree) => tree.arg)?.astId
+  } else if (tree.isTree(Tree.Type.Ident)) {
+    return tree.astId
+  }
+  return null
+})
+
+const escapeString = (str: string): string => {
+  let res = "'"
+  for (const c of str) {
+    res += c == "'" ? "\\'" : c
+  }
+  return res + "'"
+}
+const makeArgsList = (args: string[]) => '[' + args.map(escapeString).join(', ') + ']'
+
+const config = computed<Opt<NodeVisualizationConfiguration>>(() => {
+  const tree = props.input
+  const expressionId = selfArgumentAstId.value
+  const astId = tree.astId
+  if (astId == null || expressionId == null) {
+    return null
+  }
+  const info = graph.db.getMethodCallInfo(astId)
+  const args = info?.suggestion.arguments.map((arg) => arg.name)
+  const name = info?.suggestion.name
+  if (name && args) {
```

Comment:
I'd reverse this `if` to make it an early return and avoid nesting. Also, since both are based on `info` existing, it can return even earlier and avoid some null chains.
```ts
if (!info) return
```